File: Codeforces/div736/C.cpp
/**
 *    author:  kostya
 *    created: 2021-08-01 18:13:03
 *    modified 2021-08-01 18:52:24
 **/

#include <bits/stdc++.h>
using namespace::std;
#define ll long long

int main()
{
	int d;
	int m, n;
	int u, v;
	int cnt = 0;
	cin >> n >> m;
	vector<set<int>>inc(n);
	for (int i=0;i<m;++i) {
		cin >> u >> v;
		if (u > v)
			swap(u, v);
		if (inc[u].empty())
			++cnt;
		inc[u].emplace(v);
	}
	cin >> m;
	for (int i=0;i<m;++i) {
		cin >> d;
		if (d == 3)
			cout << n - cnt << endl;
		else {
			cin >> u >> v;
			if (u > v)
				swap(u, v);
			if (d == 1) {
				if (inc[u].empty())
					++cnt;
				inc[u].emplace(v);
			} else {
				inc[u].erase(v);
				if (inc[u].empty())
					--cnt;
			}
		}
	}
	return 0;
}

File: Codeforces/div736/C.py
##
#    author:  kostya
#    created: 2021-08-01 17:28:05
#    modified 2021-08-01 18:38:39
##

n, m = map(int, input().split())
inc = [set() for i in range(n)]
cnt = 0
for i in range(m):
	u, v = map(int, input().split())
	u, v = min(u, v), max(u, v)
	if len(inc[u]) == 0:
		cnt += 1
	inc[u].update((v,))
m = int(input())
for i in range(m):
	d = input().split()
	if d[0] == '3':
		print(n - cnt)
	else:
		u, v = map(int, d[1:])
		u, v = min(u, v), max(u, v)
		if d[0] == '1':
			if len(inc[u]) == 0:
				cnt += 1
			inc[u].update((v,))
		else:
			inc[u].remove(v)
			if len(inc[u]) == 0:
				cnt -= 1
